### PR TITLE
Uhm 6370 add new fields to pathology export

### PR DESCRIPTION
**Pull request by dmdesimone**

### CODE REVIEW

File: configuration/pih/liquibase/sql/create_functions.sql [hunk 1]
```diff
@@ -461,6 +461,31 @@ limit 1;
...
+	RETURNS varchar(255)
+    DETERMINISTIC
+BEGIN
+    DECLARE providerName varchar(255);
+
+select CONCAT(given_name, ' ', family_name) into providerName
+from person_name pn join provider pv on pn.person_id = pv.person_id AND pn.voided = 0
+inner join encounter_provider ep on pv.provider_id = ep.provider_id and ep.voided = 0 and ep.encounter_id = _encounter_id 
+	and ep.encounter_role_id = _provider_type 
+limit 1 offset _offset_value;
```
Comment (mseaton) [inline configuration/pih/liquibase/sql/create_functions.sql:485]: I think this likely needs some sort of fixed ordering in order for this to work?

File: configuration/pih/liquibase/sql/sql_function_reference.csv [hunk 2]
```diff
@@ -38,6 +38,7 @@ encounters,hivEncounterLocationId,encounter_id,location id,"returns the encounte
 encounters,encounter_location_name ,encounter_id,location name,location name of encounter,
 encounters,encounter_parent_location_name ,encounter_id,location name,"returns the parent location of the encounter.  It looks 3 levels deep - the encounter location, that location's parent and then that location's parent.  The first location it finds without another parent is returned.",
 encounters,provider ,encounter_id,provider name,"first and last name of provider of encounter, comma separated (this limits to 1 result, but with no sorting)",
+encounter,provider_name_of_type,"encounter_id, provider_type, offset", provider name,returns the name of the nth provider (based of offset) or the specified type,
```
Comment (mseaton) [inline configuration/pih/liquibase/sql/sql_function_reference.csv:41]: encounter -> encounters

File: configuration/reports/reportdescriptors/dataexports/sql/pathology.sql [hunk 3]
```diff
@@ -1,141 +1,208 @@
+-- set @startDate = '2022-03-28';
```
Comment (mseaton) [inline configuration/reports/reportdescriptors/dataexports/sql/pathology.sql:1]: The rest of this is impossible to review for diffs.  Not looking in detail.  Let me know if anything in particular needs a look prior to testing.

Comment (dmdesimone) [reply configuration/reports/reportdescriptors/dataexports/sql/pathology.sql:1]: no, it should be fine.  All of this is now being generated from standard functions.

**Pull request merged**